src/main/java/bio/terra/service/parquet/ParquetReaderWriterWithAvro.java

```diff
@@ -0,0 +1,151 @@
+package bio.terra.service.parquet;
+
+import bio.terra.model.TableDataType;
+import com.azure.storage.blob.BlobUrlParts;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.avro.AvroParquetReader;
+import org.apache.parquet.avro.AvroParquetWriter;
+import org.apache.parquet.hadoop.ParquetFileWriter;
+import org.apache.parquet.hadoop.ParquetReader;
+import org.apache.parquet.hadoop.ParquetWriter;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.apache.parquet.hadoop.util.HadoopInputFile;
+import org.apache.parquet.io.InputFile;
+import org.apache.parquet.io.OutputFile;
+
+public class ParquetReaderWriterWithAvro {
+
+  private ParquetReaderWriterWithAvro() {}
+
+  public static List<GenericData.Record> readFromParquet(InputFile inputFile) throws IOException {
+    List<GenericData.Record> records = new ArrayList<>();
+    try (ParquetReader<GenericData.Record> reader =
+        AvroParquetReader.<GenericData.Record>builder(inputFile).build()) {
+      GenericData.Record record;
+
+      while ((record = reader.read()) != null) {
+        records.add(record);
+      }
+      return records;
+    }
+  }
+
+  public static List<GenericData.Record> readFromParquet(Path filePath) throws IOException {
+    var config = new Configuration();
+    config.set("parquet.avro.readInt96AsFixed", "true");
+    InputFile inputFile = HadoopInputFile.fromPath(filePath, config);
+    return readFromParquet(inputFile);
+  }
+
+  public static void writeToParquet(
+      List<GenericData.Record> recordsToWrite,
+      List<String> columnNames,
+      Map<String, TableDataType> columnDataTypeMap,
+      OutputFile fileToWrite,
+      Schema schema,
+      Configuration config)
+      throws IOException {
+    try (ParquetWriter<GenericData.Record> writer =
```

Comment:
It looks like you could open both a writer and reader here and then stream the data without having to read it all into memory, is that what you were thinking we'd do?

Reply:
That's a great point - I did not think of it! I just made this improvement to the e2e example.